src/components/app/details/AboutAppInfoModal.tsx

```diff
@@ -98,60 +92,157 @@ export default function AboutAppInfoModal({
         )
     }
 
-    const handleSaveAction = (e) => {
+    const setAppTagLabel = () => {
+        const newTag = labelTags.inputTagValue.split(',').map((e) => {
+            e = e.trim()
+            return createOption(e)
+        })
+        setLabelTags({
+            inputTagValue: '',
+            tags: [...labelTags.tags, ...newTag],
+            tagError: '',
+        })
+    }
+
+    function validateForm(): boolean {
+        if (
+            labelTags.tags.length !== labelTags.tags.map((tag) => tag.value).filter((tag) => validateTags(tag)).length
+        ) {
+            setLabelTags((labelTags) => ({ ...labelTags, tagError: TAG_VALIDATION_MESSAGE.error }))
+            return false
+        }
+        return true
+    }
+
+    const handleKeyDownEvent = (e) => {
+        handleKeyDown(labelTags, setAppTagLabel, e)
+    }
+
+    function handleInputChange(inputTagValue) {
+        setLabelTags((tags) => ({ ...tags, inputTagValue: inputTagValue, tagError: '' }))
+    }
+
+    function handleTagsChange(newValue: any, actionMeta: any) {
+        setLabelTags((tags) => ({ ...tags, tags: newValue || [], tagError: '' }))
+    }
+
+    function handleCreatableBlur(e) {
+        labelTags.inputTagValue = labelTags.inputTagValue.trim()
+        if (!labelTags.inputTagValue) {
+            return
+        } else {
+            setLabelTags({
+                inputTagValue: '',
+                tags: [...labelTags.tags, createOption(e.target.value)],
+                tagError: '',
+            })
+        }
+    }
+
+    const handleSaveAction = async (e) => {
         e.preventDefault()
-        handleSubmit(e, selectedProject)
+        const validForm = validateForm()
+        if (!validForm) {
```

Comment:
we can directly use validateForm(). no need to create variable

Reply:
Done